cmd/iq_test.go

```diff
@@ -0,0 +1,273 @@
+//
+// Copyright 2018-present Sonatype Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/sonatype-nexus-community/go-sona-types/iq"
+	"github.com/sonatype-nexus-community/nancy/configuration"
+	"github.com/sonatype-nexus-community/nancy/customerrors"
+	"github.com/sonatype-nexus-community/nancy/types"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+)
+
+func TestIqApplicationFlagMissing(t *testing.T) {
+	output, err := executeCommand(rootCmd, "iq")
+	checkStringContains(t, output, "Error: required flag(s) \""+flagNameIqApplication+"\" not set")
+	assert.NotNil(t, err)
+	checkStringContains(t, err.Error(), "required flag(s) \""+flagNameIqApplication+"\" not set")
+}
+
+func TestIqHelp(t *testing.T) {
+	output, err := executeCommand(rootCmd, "iq", "--help")
+	checkStringContains(t, output, "go list -m -json all | nancy iq --"+flagNameIqApplication+" your_public_application_id --"+flagNameIqServerUrl+" ")
+	assert.Nil(t, err)
+}
+
+func setupIQConfigFile(t *testing.T, tempDir string) {
+	cfgDirIQ := path.Join(tempDir, types.IQServerDirName)
+	assert.Nil(t, os.Mkdir(cfgDirIQ, 0700))
+
+	cfgFileIQ = path.Join(tempDir, types.IQServerDirName, types.IQServerConfigFileName)
+}
+func resetIQConfigFile() {
+	cfgFileIQ = ""
+}
+
+func TestInitIQConfig(t *testing.T) {
+	viper.Reset()
+	defer viper.Reset()
+
+	tempDir := setupConfig(t)
+	defer resetConfig(t, tempDir)
+
+	setupTestOSSIConfigFileValues(t, tempDir)
+	defer func() {
+		resetOSSIConfigFile()
+	}()
+
+	setupIQConfigFile(t, tempDir)
+	defer func() {
+		resetIQConfigFile()
+	}()
+
+	const credentials = configuration.YamlKeyIQUsername + ": iqUsernameValue\n" +
```

Comment:
You could use `fmt.Sprintf("%s: %s\n", configuration.YamlKeyIQUsername, iqUsernameValue)` etc... to maybe make this read easier.

Reply:
> You could use `fmt.Sprintf("%s: %s\n", configuration.YamlKeyIQUsername, iqUsernameValue)` etc... to maybe make this read easier.

Cool. Bring more brain train. Done.  0b87905